app/controllers/PasskeyController.scala

```diff
@@ -83,50 +86,71 @@ class PasskeyController(
+          case "error.required"  => "missing value"
+          case "error.maxLength" => "too long"
+          case "error.pattern"   => "contains invalid characters"
+          case other             => other
+        }
+        s"${error.key}: $message"
+      }
+      .mkString(", ")
+
   /** See
     * [[https://webauthn4j.github.io/webauthn4j/en/#registering-the-webauthn-public-key-credential-on-the-server]].
     */
   def register: Action[Map[String, Seq[String]]] = authAction(
     parse.formUrlEncoded
-  ) { request =>
-    apiResponse(
-      for {
-        challengeResponse <- PasskeyChallengeDB.loadChallenge(request.user)
-        challenge <- PasskeyChallengeDB.extractChallenge(
-          challengeResponse,
-          request.user
-        )
-        passkey <- request.body.get("passkey") match {
-          case Some(values) => Success(values.head)
-          case None =>
-            Failure(
-              JanusException.missingFieldInRequest(request.user, "passkey")
+  ) { implicit request =>
+    registrationForm
+      .bindFromRequest()
+      .fold(
+        formWithErrors =>
+          Redirect("/user-account")
+            .flashing("error" -> formattedErrors(formWithErrors)),
+        registrationData =>
+          apiResponse(for {
+            challengeResponse <- PasskeyChallengeDB.loadChallenge(request.user)
+            challenge <- PasskeyChallengeDB.extractChallenge(
+              challengeResponse,
+              request.user
             )
-        }
-        passkeyName <- request.body.get("passkeyName") match {
-          case Some(values) => Success(values.head)
-          case None =>
-            Failure(
-              JanusException.missingFieldInRequest(request.user, "passkeyName")
+            credRecord <- Passkey.verifiedRegistration(
+              host,
+              request.user,
+              challenge,
+              registrationData.passkey
             )
-        }
-        credRecord <- Passkey.verifiedRegistration(
-          host,
-          request.user,
-          challenge,
-          passkey
-        )
-        _ <- PasskeyDB.insert(request.user, credRecord, passkeyName)
-        _ <- PasskeyChallengeDB.delete(request.user)
-        _ = logger.info(s"Registered passkey for user ${request.user.username}")
-      } yield {
-        val message =
-          if (passkeyName.nonEmpty)
-            s"Passkey '$passkeyName' was registered successfully"
-          else "Passkey was registered successfully"
-        Redirect("/user-account").flashing("success" -> message)
-      }
-    )
+            _ <- PasskeyDB
+              .insert(request.user, credRecord, registrationData.passkeyName)
+            _ <- PasskeyChallengeDB.delete(request.user)
+            _ = logger
+              .info(s"Registered passkey for user ${request.user.username}")
+          } yield {
+            Redirect("/user-account")
+              .flashing("success" -> "Passkey was registered successfully")
```

Comment:
Any reason why we are no longer showing the passkey name in the message?

Reply:
No, just an oversight.  Thanks for spotting!